app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.connectors
+
+import java.time.LocalDate
+
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.merchandiseinbaggage.config.CurrencyConversionConfiguration
+import uk.gov.hmrc.merchandiseinbaggage.model.api.CurrencyConversionResponse
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CurrencyConversionConnector extends CurrencyConversionConfiguration {
+
+  def findCurrencyConversion(httpClient: HttpClient, currencyCode: String, date: LocalDate)
```

Comment:
What do we gain from using this pattern?

I assume we need to inject a httpClient at controller level and then extend this trait, then pass the injected httpClient to any calls to this method, seems a bit messy.

I would prefer this was a Singleton class with httpClient injected, then this gets injected into controllers rather than extended.

If you prefer to have it like this for easy unit test then maybe declare val httpClient outside of the method call then when we inject httpClient at controller level we can use override modifier, and that will mean we won't need to explicitly pass httpClient to this function every time we need it?

Reply:
_What do we gain from using this pattern?_   don't need to Guice it up so cleaner an match easier to test but of course I wish we could get the httpClient mixed in from an other trait, that will really be the nice way of doing it but.........